### DISCUSSION TITLE
TLS Client Authentication

### DISCUSSION THREAD

**Discussion by vidarne** [Ylianst/MeshCentral#4048] [Ideas]
Reverse Proxy

Any plans for the future that the MeschCentral Agent/MeschCentral Router will get option to autenticate whit TLS Client Authentication (User certificate)?

So that it is posible that only clients with a valid certificate can communicate with the MeschCentral backend via a reverse Proxy?

Did see there is the posible to split web port and client port  whit "agentport" and then use TLS client certs for the web portal, but an option to use client certs for autenticating the Agent/Router be greate for exstra securety.

**Comment by Ylianst** [comment]
This is an interesting request. You can add it as an issue / feature request for sure. I had not considered this one, obviously it would only work on a "split port" and would need a bunch of changes to the tools to do this. It would indeed provide much better security since calls would be blocked at the TLS layer.
